From d068e0db3eff23e8d129de0af7445edc409eedb6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb@kaidan.im>
Date: Sat, 4 Mar 2023 02:46:47 +0100
Subject: [PATCH] Add fallback theming

---
 src/main.cpp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 8542935f..244ac8db 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -7,11 +7,13 @@
 #include <QtQml>
 #include <QSortFilterProxyModel>
 #include <QUrl>
+#include <QQuickWindow>
+#include <QQuickStyle>
+
 #include <KLocalizedContext>
 #include <KLocalizedString>
 #include <KCrash>
 #include <KAboutData>
-#include <QQuickWindow>
 
 #include "searchmodel.h"
 #include "albummodel.h"
@@ -31,6 +33,12 @@ constexpr auto URI = "org.kde.ytmusic";
 
 Q_DECL_EXPORT int main(int argc, char *argv[])
 {
+    // set default style and icon theme
+    QIcon::setFallbackThemeName(QStringLiteral("breeze"));
+    if (qEnvironmentVariableIsEmpty("QT_QUICK_CONTROLS_STYLE") && QQuickStyle::name().isEmpty()) {
+        QQuickStyle::setStyle(QStringLiteral("org.kde.desktop"));
+    }
+
     QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
     QApplication app(argc, argv);
     QCoreApplication::setOrganizationName(QStringLiteral("KDE"));
-- 
GitLab

