From 78d033990bc42a2d4faed66c17a20898d239185a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb@kaidan.im>
Date: Fri, 29 May 2026 13:08:07 +0200
Subject: [PATCH] Add video url fallback

+ debug logging
---
 src/videoinfoextractor.cpp | 25 +++++++++++++++++++------
 1 file changed, 19 insertions(+), 6 deletions(-)

diff --git a/src/videoinfoextractor.cpp b/src/videoinfoextractor.cpp
index b24cf0d0..81d445ee 100644
--- a/src/videoinfoextractor.cpp
+++ b/src/videoinfoextractor.cpp
@@ -38,21 +38,34 @@ QUrl VideoInfoExtractor::audioUrl() const
 
     std::vector<video_info::Format> audioFormats;
 
+    qWarning() << "available formats:";
+    for (const auto &f : m_videoInfo.formats) {
+        qWarning() << "  " << "acodec" << f.acodec << "format_id" << f.format_id << "quality" << f.quality << "vcodec" << f.vcodec;
+    }
+
     // filter audio only formats
     std::copy_if(m_videoInfo.formats.begin(), m_videoInfo.formats.end(), std::back_inserter(audioFormats),
         [](const video_info::Format &format) {
         return (!format.acodec.has_value() || format.acodec != "none") && format.vcodec == "none";
     });
 
+    std::sort(audioFormats.begin(), audioFormats.end(), [](const video_info::Format &a, const video_info::Format &b) {
+        return a.quality > b.quality;
+    });
+
     if (audioFormats.empty()) {
-        qWarning() << "No audio track found";
-        return {};
+        qWarning() << "No audio track found, falling back to worst-quality video";
+
+        std::copy_if(m_videoInfo.formats.begin(), m_videoInfo.formats.end(), std::back_inserter(audioFormats), [](const video_info::Format &format) {
+            return format.acodec != "none";
+        });
+
+        std::sort(audioFormats.begin(), audioFormats.end(), [](const video_info::Format &a, const video_info::Format &b) {
+            return a.quality < b.quality;
+        });
     }
 
-    std::sort(audioFormats.begin(), audioFormats.end(),
-              [](const video_info::Format &a, const video_info::Format &b) {
-        return a.quality > b.quality;
-    });
+    qWarning() << "chosen url:" << audioFormats.front().url;
 
     return QUrl(QString::fromStdString(audioFormats.front().url));
 }
-- 
2.54.0

