From 219d5462f6995e27e85969a8ea73a3c1b13841fb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb@kaidan.im>
Date: Sat, 22 Apr 2023 13:19:19 +0200
Subject: [PATCH 5/5] If controls style is not set, assume platform provides no
 working styling

---
 src/main.cpp | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index 2826d2e..dfd9f4f 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -9,6 +9,8 @@
 #include <QUrl>
 #include <QQuickWindow>
 #include <QQuickStyle>
+#include <QStyle>
+#include <QStyleFactory>
 
 #include <KLocalizedContext>
 #include <KLocalizedString>
@@ -36,14 +38,17 @@ constexpr auto URI = "org.kde.ytmusic";
 
 Q_DECL_EXPORT int main(int argc, char *argv[])
 {
+    QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+    QApplication app(argc, argv);
+
     // set default style and icon theme
     QIcon::setFallbackThemeName(QStringLiteral("breeze"));
     if (qEnvironmentVariableIsEmpty("QT_QUICK_CONTROLS_STYLE") && QQuickStyle::name().isEmpty()) {
         QQuickStyle::setStyle(QStringLiteral("org.kde.desktop"));
+        QApplication::setStyle(QStyleFactory::create(QStringLiteral("Breeze")));
     }
 
-    QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
-    QApplication app(argc, argv);
+
     QCoreApplication::setOrganizationName(QStringLiteral("KDE"));
     QCoreApplication::setOrganizationDomain(QStringLiteral("kde.org"));
     QCoreApplication::setApplicationName(QStringLiteral("audiotube"));
-- 
2.40.0

