From df0cb9f43c7abbe132ddb04789cfbb2affaea8b0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonah=20Br=C3=BCchert?= <jbb@kaidan.im>
Date: Sat, 20 May 2023 18:13:49 +0200
Subject: [PATCH] Make media duration key optional

---
 src/ytmusic.cpp | 4 ++--
 src/ytmusic.h   | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ytmusic.cpp b/src/ytmusic.cpp
index 14de95a3..476602d8 100644
--- a/src/ytmusic.cpp
+++ b/src/ytmusic.cpp
@@ -292,7 +292,7 @@ std::optional<search::SearchResultItem> extract_search_result(py::handle result)
                 result["videoId"].cast<std::string>(),
                 result["title"].cast<std::string>(),
                 extract_py_list<meta::Artist>(result["artists"]),
-                result["duration"].cast<std::string>(),
+                optional_key<std::string>(result, "duration"),
                 extract_py_list<meta::Thumbnail>(result["thumbnails"])
             },
             optional_key<std::string>(result, "views")
@@ -303,7 +303,7 @@ std::optional<search::SearchResultItem> extract_search_result(py::handle result)
                 result["videoId"].cast<std::string>(),
                 result["title"].cast<std::string>(),
                 extract_py_list<meta::Artist>(result["artists"]),
-                result["duration"].cast<std::string>(),
+                optional_key<std::string>(result, "duration"),
                 extract_py_list<meta::Thumbnail>(result["thumbnails"])
             },
             result["album"].is_none() ? std::optional<meta::Album> {} : extract_meta_album(result["album"]),
diff --git a/src/ytmusic.h b/src/ytmusic.h
index f8768571..b0209e59 100644
--- a/src/ytmusic.h
+++ b/src/ytmusic.h
@@ -45,7 +45,7 @@ struct Media {
     std::string video_id;
     std::string title;
     std::vector<meta::Artist> artists;
-    std::string duration;
+    std::optional<std::string> duration;
     std::vector<meta::Thumbnail> thumbnails;
 };
 
-- 
GitLab

